otus-kuber-2019-12/ifqthenp_platform
# HW1. Kubernetes Intro

# Выполнено ДЗ №1

 - [x] Основное ДЗ
 - [x] Задание со *

## В процессе сделано:
- Created Docker image with simple http server using `kubernetes-intro/web/Dockerfile` file
- Created Kubernetes manifest file describing Pod with `containers` and `initContainer` specs
- Fixed broken `frontend` Pod from [Hipster Shop repo][1] and saved result in `kubernetes-intro/frontend-pod-healthy.yaml`

[1]: https://github.com/GoogleCloudPlatform/microservices-demo

## Как запустить проект:
 - run `kubectl apply -f kubernetes-intro/web-pod.yaml`
 - run `kubectl apply -f kubernetes-intro/frontend-pod-healthy.yaml`

## Как проверить работоспособность:
 - to test that `web` pod is running:
     - run `kubectl port-forward --address 0.0.0.0 pod/web 8000:8000`
     - open in browser `http://localhost:8000/index.html`
 - to test that `frontend` microservice is running without errors:
     - run `kubectl get pods` and make sure that the `STATUS` of the `frontend` pod is `Running`

## PR checklist:
 - [x] Выставлен label с номером домашнего задания
